File: cashChange.swift

import Foundation

func findMinimumNotes(_ N: Int){
    let denominations = [500, 100, 50, 10, 5, 1]
    var amount = N
    var result: [Int] = []

    for note in denominations{
        while amount >= note{
            amount -= note
            result.append(note)
        }
    }
    result.sort()
    print(result.map{String($0)}.joined(separator: " "))
}

if let input = readLine(), let N = Int(input){
    findMinimumNotes(N)
}

File: cashChange.cpp
#include<bits/stdc++.h>
using namespace std;


void findMinimumNotes(int N){
    vector<int> denominations = {500, 100, 50, 10, 5, 1};
    vector <int> result;

    for(int  note: denominations){
        while(N>=note){
            N -= note;
            result.push_back(note);
        }
    }

    sort(result.begin(), result.end());
    for(int note : result){
        cout<<note<<" ";
    }
    cout<<endl;
}


int main()
{
int N;
cin>>N;

findMinimumNotes(N);

    return 0;
}